design/garbage-collection.md

```diff
@@ -39,18 +77,85 @@ In the initial state, both clients have `"ab"`.
 
 ![garbage-collection-2](media/garbage-collection-2.png)
 
-Client `c1` deletes `"b"`, which is recorded as a change with logical timestamp `3`. The text node of `"b"` can be referenced by remote, so it is only marked as tombstone. And the client `c1` sends change `3` to server through PushPull API and receives as a response that `min_synced_seq` is `2`. Since all clients did not receive the deletion `change 3`, the text node is not purged by garbage collection.
+`Client a` deletes `"b"`, which is recorded as a change with versionVector `{b:1}`. The text node of `"b"` can be referenced by remote, so it is only marked as tombstone. And the `Client a` sends change `3a` to server through PushPull API and receives as a response that `minVersionVector` is `{a:1, b:2}`. Since all clients did not receive the deletion `change 3a`, the text node is not purged by garbage collection.
 
-Meanwhile, client `c2` inserts `"c"` after textnode `"b"`.
+Meanwhile, `client b` inserts `"c"` after textnode `"b"`.
 
 #### State 3
 
 ![garbage-collection-3](media/garbage-collection-3.png)
 
-Client `c2` pushes change `4` to server and receives as a response that `min_synced_seq` is `3`. After the client applies change `4`, the contents of document are changed to `ac`. This time, all clients have received change `3`, so textnode `"b"` is completely removed.
+`Client b` pushes change `3b` to server and receives as a response that `minVersionVector` is `{a:1, b:1}`. After the client applies change `4`, the contents of document are changed to `ac`. This time, all clients have received change `3a`. Since node "b" is removed at `3a`, it's still marked as tombstone for every clients, because `minVersionVector[a] = 1 < 3` 
 
 #### State 4
 
 ![garbage-collection-4](media/garbage-collection-4.png)
 
-Finally, after client `c1` receives change `4` from server, purges textnode `"b"` because it is no longer referenced remotely.
+`Client a` pushpull but nothing to push or pull. `minVersionVector` is still `{a:1, b:1}`, so no GC happens.
+
+#### State 5
+
+![garbage-collection-4](media/garbage-collection-5.png)
+
+`Client b` pushpull but nothing to push or pull. `minVersionVector` is now `{a:3, b:1}`, node "b" removedAt is `3@a`, and `minVersionVector[a] = 3 >= 3`, thus `client b` meets GC condition 
```

Comment:
why `client b` pull `minVersionVector{a:3, b:1}`? Above image `garbage-collection-4.png`, `client a` pull `minVersionVector{a:1, b:1}`.

Reply:
Ah my bad, I'll fix it 
Thanks for your correction